### PR TITLE
feat: Add WeeklyConfigBackupFeature

### PR DESCRIPTION
**Pull request by kristofbolyai**
This should prevent accidental data loss, if something happens (no matter if we were the mod crashing or not). Some mods crash on tick, causing us to leave open files on windows, deleting them.

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/utilities/WeeklyConfigBackupFeature.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Persisted
+    private final Storage<Long> lastBackup = new Storage<>(0L);
+
+    @SubscribeEvent
+    public void onTitleScreenInit(TitleScreenInitEvent.Post event) {
+        long currentTime = System.currentTimeMillis();
+        long lastBackupTime = lastBackup.get();
+
+        if (currentTime - lastBackupTime >= BACKUP_INTERVAL_DAYS * 24 * 60 * 60 * 1000) {
```
Comment (ryanzhoudev) [inline common/src/main/java/com/wynntils/features/utilities/WeeklyConfigBackupFeature.java]: TimeUnit.DAYS.toMillis() could be cleaner here

File: common/src/main/java/com/wynntils/features/utilities/WeeklyConfigBackupFeature.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import com.wynntils.core.persisted.config.Category;
+import com.wynntils.core.persisted.config.ConfigCategory;
+import com.wynntils.core.persisted.storage.Storage;
+import com.wynntils.mc.event.TitleScreenInitEvent;
+import com.wynntils.utils.FileUtils;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.UTILITIES)
```
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/features/utilities/WeeklyConfigBackupFeature.java]: Maybe the Wynntils category?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/utilities/WeeklyConfigBackupFeature.java]: Sure, although it fits both.

**Review by ShadowCat117 — APPROVED** [review]

File: common/src/main/java/com/wynntils/features/utilities/WeeklyConfigBackupFeature.java (hunk 2 above)
Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/features/utilities/WeeklyConfigBackupFeature.java]: Would be nice if we could add multiple categories, I think someone has mentioned that before.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/utilities/WeeklyConfigBackupFeature.java]: magicus did, we mostly need UI refactors for that.

**Pull request merged**